lxd/instance/drivers/driver_qemu.go

```diff
@@ -5777,12 +5777,18 @@ func (d *qemu) writeInstanceData() error {
 		location = d.Location()
 	}
 
+	id := d.LocalConfig()["volatile.cloud-init.instance-id"]
+	if id == "" {
+		id = d.Name()
+	}
+
 	out, err := json.Marshal(struct {
 		Name     string                         `json:"name"`
+		ID       string                         `json:"id"`
```

Comment:
Canwe call this field `CloudInitID` as otherwise it could easily be confused with the instance's actual `ID()` function/field?